BuildResidentialHPXML/measure.rb

```diff
@@ -3318,20 +3345,63 @@ def self.set_air_infiltration_measurements(hpxml, runner, args)
   def self.set_attics(hpxml, runner, model, args)
     return if args[:geometry_unit_type] == HPXML::ResidentialTypeApartment
 
+    surf_ids = { 'roofs' => { 'surfaces' => hpxml.roofs, 'ids' => [] },
+                 'walls' => { 'surfaces' => hpxml.walls, 'ids' => [] },
+                 'frame_floors' => { 'surfaces' => hpxml.frame_floors, 'ids' => [] } }
+
+    attic_locations = [HPXML::LocationAtticUnconditioned, HPXML::LocationAtticUnvented, HPXML::LocationAtticVented]
+    surf_ids.each do |surf_type, surf_hash|
+      surf_hash['surfaces'].each do |surface|
+        next if (not attic_locations.include? surface.interior_adjacent_to) &&
+                (not attic_locations.include? surface.exterior_adjacent_to)
+
+        surf_hash['ids'] << surface.id
+      end
+    end
+
     if args[:geometry_roof_type] == 'flat'
       hpxml.attics.add(id: HPXML::AtticTypeFlatRoof,
                        attic_type: HPXML::AtticTypeFlatRoof)
     else
       hpxml.attics.add(id: args[:geometry_attic_type],
-                       attic_type: args[:geometry_attic_type])
+                       attic_type: args[:geometry_attic_type],
+                       attached_to_roof_idrefs: surf_ids['roofs']['ids'],
+                       attached_to_wall_idrefs: surf_ids['walls']['ids'],
+                       attached_to_frame_floor_idrefs: surf_ids['frame_floors']['ids'])
     end
   end
 
   def self.set_foundations(hpxml, runner, model, args)
     return if args[:geometry_unit_type] == HPXML::ResidentialTypeApartment
 
+    surf_ids = { 'slabs' => { 'surfaces' => hpxml.slabs, 'ids' => [] },
+                 'frame_floors' => { 'surfaces' => hpxml.frame_floors, 'ids' => [] },
+                 'foundation_walls' => { 'surfaces' => hpxml.foundation_walls, 'ids' => [] },
+                 'walls' => { 'surfaces' => hpxml.walls, 'ids' => [] },
+                 'rim_joists' => { 'surfaces' => hpxml.rim_joists, 'ids' => [] }, }
+
+    foundation_locations = [HPXML::LocationBasementConditioned, HPXML::LocationBasementUnconditioned,
+                            HPXML::LocationCrawlspaceUnvented, HPXML::LocationCrawlspaceVented,
+                            HPXML::FoundationTypeAmbient]
```

Comment:
Since this is a HPXML::Foundation enum and not a HPXML::Location enum, I have a hard time believing this works correctly for ambient foundations. Have you double-checked this?

Reply:
I added because of this comment: https://github.com/NREL/resstock/pull/668#discussion_r681331384, but what surfaces are ever attached to ambient foundations? Should I remove the foundation type or is there an ambient location to check?

When I run a building with ambient there are no surfaces adjacent to the foundation

Reply:
We might have to get @nmerket to weigh in here. As far as I can tell, ambient foundations [are treated as crawlspaces](https://hescore-hpxml.readthedocs.io/en/latest/translation/zone_floor.html#foundation-type) in hescore-hpxml, which means that it's going to look for foundation walls and a floor above. We have a floor above, which you should be able to update this code to handle, but we don't have any foundation walls -- I don't know what the hpxml-hescore translator will do if it finds this scenario.